processing/SpaceCrashAlpha/PhysicsObject.pde

```diff
@@ -0,0 +1,93 @@
+import java.util.Iterator;
+
+class PhysicsObjects<T extends PhysicsObject> extends ArrayList<T> {
+  void update() {
+    Iterator iter = this.iterator();
+    while (iter.hasNext()) {
+      T t = (T) iter.next();
+      t.update();
+    }
+  }
+}
+
+abstract class PhysicsObject {
+  PVector acceleration;
+  PVector velocity;
+  PVector position;
+  PVector lastPosition;
+  float mass;
+  boolean immovable; // can this thing be moved by collisions?
+  float drag;
+  
+  void update() {
+    if (!immovable) {
+      lastPosition = position.copy();
+      velocity.add(acceleration);
+      velocity.mult(1.0 - drag);
+      position.add(velocity);
+      acceleration.set(0,0);
+    }
+  }
+  
+  PVector getPosition() {
+    return position;
+  }
+  void setPosition(int newX, int newY) {
+    position.set(newX, newY);
+  }
+  void setPosition(PVector newPosition) {
+    position = newPosition;
+  }
+  
+  PVector getLastPosition() {
+    return lastPosition;
+  }
+  void setLastPosition(int newX, int newY) {
+    lastPosition.set(newX, newY);
+  }
+  void setLastPosition(PVector newPosition) {
+    lastPosition = newPosition;
+  }
+  
+  PVector getVelocity() {
+    return velocity;
+  }
+  void setVelocity(int newX, int newY) {
+    velocity.set(newX, newY);
+  }
+  void setVelocity(PVector newVelocity) {
+    velocity = newVelocity;
+  }
+  
+  void setDrag(float newDrag) {
+    drag = newDrag;
+  }
+  
+  void applyForce(PVector force) {
+    PVector accToAdd = PVector.div(force, mass);
+    acceleration.add(accToAdd);
+  }
+}
+
+class PointMass extends PhysicsObject {
+  float radius;
+  
+  PointMass(boolean isImmovable) {   
```

Comment:
this seems like a weird and arbitrary parameter to pass into the constructor. but it feels even weirder to have a default setting and then call something like `setImmovable()` on it afterwards. Java doesn't seem to support default values for parameters, but is there a reasonable solution, maybe by overloading the constructor?